truffleruby/src/main/java/org/truffleruby/language/threadlocal/GetFromThreadLocalNode.java

```diff
@@ -31,14 +34,13 @@ public Object isDefined(VirtualFrame frame) {
     public Object execute(VirtualFrame frame) {
         final Object threadLocalObject = value.execute(frame);
 
-        if (RubyGuards.isThreadLocal(threadLocalObject)) {
+        if (isThreadLocalProfile.profile(RubyGuards.isThreadLocal(threadLocalObject))) {
             return getThreadLocalValue((ThreadLocalObject) threadLocalObject);
         }
 
         return threadLocalObject;
     }
 
-    @TruffleBoundary
     private Object getThreadLocalValue(ThreadLocalObject threadLocalObject) {
```

Comment:
`ThreadLocal.get` is too big for partial evaluation, so it should remain a boundary.